src/app/rpmostree-builtin-deploy.cxx

```diff
@@ -119,6 +123,32 @@ rpmostree_builtin_deploy (int            argc,
     }
   else
     {
+      if (!opt_bypass_driver)
```

Comment:
So my general rule for copy/paste is that having similar code in two places isn't the end of the world - ideally they're at least cross referenced with something like comment "// if changing this function foo, also change bar()".

But if you have the same code in *three* (or more) places it really demands refactoring to share things.  I think this one should be quite easy to share as a common helper function in e.g. `rpmostree-dbus-helpers.cxx`.

Reply:
It may be simplest to just say "and perform the operation anyways" instead of the two cases of "and deploy anyway" and "and rebase anyway".

Reply:
> But if you have the same code in three (or more) places it really demands refactoring to share things.

Sorry for that, I also felt like I was really pushing it when I submitted the code (no pun intended). I originally envisioned that the error messages would have more variation, so I only factored out the parts that I thought would 100% be identical. But I think your suggestion of making the error message slightly more generalized would make everything much cleaner.

I've refactored it properly now :)